src/libraries/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -136,42 +140,84 @@ public unsafe BitArray(bool[] values)
 
             if (Avx2.IsSupported)
             {
+                // JIT does not support code hoisting for SIMD yet
+                Vector256<byte> zero = Vector256<byte>.Zero;
                 fixed (bool* ptr = values)
                 {
-                    for (; (i + Vector256<byte>.Count) <= values.Length; i += Vector256<byte>.Count)
+                    for (; (i + Vector256<byte>.Count) <= values.Length; i += (uint)Vector256<byte>.Count)
```

Comment:
is `(i + Vector256<byte>.Count)` implicitly casting to `long`/`ulong`? The same for the comparison against `values.Length`?

Reply:
I really hope I haven't missed one 🙄. Should be fixed now.